src/demo/DemoTestRepaymentService.java

```diff
@@ -215,4 +212,126 @@ public static void testGetInvestorAccountRepayments() throws MambuApiException {
+
+			List<Repayment> returnedRepayments = submitEditedRepaymentsToMambu(notPaidRepayments);
+
+			// delete first of the repayments
+			RepaymentsService repaymentService = MambuAPIFactory.getRepaymentsService();
+			Boolean deleted = repaymentService.deleteLoanRepayment(LOAN_ACCOUNT_ID,
+					returnedRepayments.get(0).getEncodedKey());
+
+			System.out.println("Repayment was successfully deleted = " + deleted);
+		} else {
+			System.out.println(methodName + "can`t be performed due e to repayments no.");
+			System.out.println("There should be at least 2 repayments");
+		}
+	}
+
+	/**
+	 * Submits the list of updated repayments received as parameter to this method in order to update the schedule of a
+	 * loan.
+	 * 
+	 * @param editedRepayments
+	 *            a list of updated repayments to be sent to Mambu.
+	 * 
+	 * @return a list of updated repayments from Mambu
+	 * 
+	 * @throws MambuApiException
+	 */
+	private static List<Repayment> submitEditedRepaymentsToMambu(List<Repayment> editedRepayments)
+			throws MambuApiException {
+
+		// Submit Update schedule API request with these updated entries
+		JSONLoanRepayments loanRepayments = new JSONLoanRepayments(editedRepayments);
+		RepaymentsService repaymentService = MambuAPIFactory.getRepaymentsService();
+		System.out.println("Updating Repayments schedule for Loan Account ID=" + LOAN_ACCOUNT_ID + "\tRepayments ="
+				+ editedRepayments.size());
+
+		List<Repayment> updatedRepayments = repaymentService.updateLoanRepaymentsSchedule(LOAN_ACCOUNT_ID,
+				loanRepayments);
+
+		int totalReturned = (updatedRepayments == null) ? 0 : updatedRepayments.size();
+		System.out.println("Total Repayments returned after update=" + totalReturned);
+		// Can also see detailed update log on a Dashboard in Mambu
+		return updatedRepayments;
+	}
+
+	/**
+	 * Edits the list of not paid yet repayments by setting the first one`s principal to be "0"
+	 * 
+	 * @param notPaidRepayments
+	 *            the list of not paid repayments to be adjusted.
+	 */
+	private static void reduceScheduleByFirstRepayment(List<Repayment> notPaidRepayments) {
+
+		BigDecimal newInstallementsNo = new BigDecimal(notPaidRepayments.size() - 1);
+
+		Money totalPrincipalAmountBalance = demoLoanAccount.getPrincipalAmount();
+
+		// calculate the due for remaining repayments
+		BigDecimal repaymentPrincipal = new BigDecimal(totalPrincipalAmountBalance.getAmount()
+				.divide(newInstallementsNo, 2, RoundingMode.HALF_UP).doubleValue());
+
+		Money principal = new Money(repaymentPrincipal.doubleValue());
+		principal.setScale(2, RoundingMode.HALF_UP);
+
+		// update repayments (all but first)
+		for (int i = 1; i < notPaidRepayments.size() - 1; i++) {
+			notPaidRepayments.get(i).setPrincipalDue(principal);
+		}
+
+		// set the first one to be = "0"
+		notPaidRepayments.get(0).setPrincipalDue(new BigDecimal("0"));
+
+		BigDecimal updatedRepaymentsNo = new BigDecimal(notPaidRepayments.size() - 1);
```

Comment:
Couldn't we use newInstallmentsNo's value here? It seems that it has the same value.